File: app/src/main/java/tech/pauly/findapet/data/models/Status.java
package tech.pauly.findapet.data.models;

import android.support.annotation.Nullable;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Root(name = "status")
public class Status {
    @Element
    private int code;

    @Element(required = false)
    private String message;

    public StatusCode getCode() {
        return StatusCode.Companion.fromInt(code);
    }

    @Nullable
    public String getMessage() {
        return message;
    }
}

File: app/src/main/java/tech/pauly/findapet/data/models/Status.kt
package tech.pauly.findapet.data.models

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "status", strict = false)
open class Status {

    @field:Element(name = "code")
    private var _code: Int = 0
    open val code: StatusCode
        get() = StatusCode.fromInt(_code)

    @field:Element(required = false)
    open var message: String? = null
}